app/src/lib/stores/app-store.ts

```diff
@@ -8483,6 +8502,32 @@ function isLocalChangesOverwrittenError(error: Error): boolean {
   )
 }
 
+function isPreCommitHookFailure(error: Error): boolean {
+  if (error instanceof GitError && error.result.exitCode !== 0) {
+    const output = extractHookOutput(error).toLowerCase()
+
+    // Check for pre-commit hook failure indicators (case insensitive)
+    return output.includes('husky') ||
+           output.includes('pre-commit') ||
+           output.includes('lint-staged')
+  }
+
+  return false
+}
+
+function extractHookOutput(error: Error): string {
+  if (error instanceof ErrorWithMetadata) {
+    return extractHookOutput(error.underlyingError)
+  }
```

Comment:
Looking at `isPreCommitHookFailure` that checks for `instanceof GitError` is this even possible?

I'd say this function should take a `error: GitError` and simplify its logic.